client/src/components/ItemPic/index.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import ZoomImg from './ZoomImg';
+import './ItemPic.css';
+
+class ItemPic extends Component {
+  state = {
+    currentIndex: 0,
+    translateValue: 0,
+  };
+
+  componentDidMount() {
+    window.addEventListener('resize', () => {
+      this.setState({
+        currentIndex: 0,
+        translateValue: 0,
+      });
+    });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize');
+  }
+
+  slideWidth = element => {
+    const { currentIndex } = this.state;
+    const { picArr } = this.props;
+
+    let sliderWidth;
+
+    if (element === 'current_img') {
+      sliderWidth = document.querySelector(`#img${currentIndex}`).clientWidth;
+    }
+    if (element === 'slider') {
+      sliderWidth = document.querySelector(`#img${currentIndex}`).clientWidth;
+      picArr.forEach((item, index) => {
+        sliderWidth -= document.querySelector(`#img${index}`).clientWidth;
+      });
+    }
+    return sliderWidth;
+  };
+
+  nextImg = () => {
+    const { currentIndex, translateValue } = this.state;
+    const { picArr } = this.props;
+
+    if (currentIndex === picArr.length - 1) {
+      this.setState({
+        currentIndex: 0,
+        translateValue: 0,
+      });
+    } else {
+      this.setState({
+        currentIndex: currentIndex + 1,
+        translateValue: translateValue - this.slideWidth('current_img'),
+      });
+    }
+  };
+
+  prevImg = () => {
+    const { currentIndex, translateValue } = this.state;
+    const { picArr } = this.props;
+
+    if (currentIndex === 0) {
+      this.setState({
+        currentIndex: picArr.length - 1,
+        translateValue: this.slideWidth('slider'),
+      });
+    } else {
+      this.setState({
+        currentIndex: currentIndex - 1,
+        translateValue: translateValue + this.slideWidth('current_img'),
+      });
+    }
+  };
+
+  clickedImg = e => {
+    const { currentIndex, translateValue } = this.state;
+    const translateAdd =
+      (currentIndex - Number(e.target.id)) * this.slideWidth('current_img');
+
+    this.setState({
+      currentIndex: Number(e.target.id),
+      translateValue: translateValue + translateAdd,
+    });
+  };
+
+  render() {
+    const { translateValue, currentIndex } = this.state;
+    const { picArr } = this.props;
+    return (
+      <div className="item-pic">
+        <div className="item-pic__slider">
+          <div
```

Comment:
The tag icon that shows the product status (sale, sold out) is missing here

Reply:
I added it